pyexchange/coinone.py

```diff
@@ -0,0 +1,256 @@
+                     timestamp=int(item['timestamp']),
+                     pair=pair,
+                     is_sell=True if item['type'] == 'ask' else False,
+                     price=Wad.from_number(float(item["price"])),
+                     amount=Wad.from_number(float(item['qty'])))
+
+
+class CoinoneTrade(Trade):
+
+    @staticmethod
+    def from_message(trade, pair: str) -> Trade:
+        return Trade(trade_id=trade['orderId'],
+                     timestamp=int(trade['timestamp']),
+                     pair=pair,
+                     is_sell=True if trade['type'] == 'ask' else False,
+                     price=Wad.from_number(float(trade["price"])),
+                     amount=Wad.from_number(float(trade['qty'])))
+
+
+class CoinoneApi(PyexAPI):
+    """Coinone API interface.
+
+        Documentation available here: https://doc.coinone.co.kr/
+
+        Precision requirement information available here: https://coinone.co.kr/support/guide
+    """
+
+    logger = logging.getLogger()
+
+    def __init__(self, api_server: str, access_token: str, secret_key: str, timeout: float = 9.5):
+        assert (isinstance(api_server, str))
+        assert (isinstance(access_token, str))
+        assert (isinstance(secret_key, str))
+        assert (isinstance(timeout, float))
+
+        self.api_server = api_server
+        self.access_token = access_token
+        self.secret_key = secret_key
+        self.timeout = timeout
+        self.last_nonce = 0
+        self.last_nonce_lock = threading.Lock()
+
+    def get_balances(self) -> dict:
+        return self._http_authenticated_request("POST", "/v2/account/balance", {})
+
+    # Doesn't retrieve precision information for the market, only orderbook info
+    def get_markets(self) -> List:
+        return self._http_unauthenticated_request("GET", "/orderbook", {})
+
+    def get_pair(self, pair: str) -> List:
+        assert (isinstance(pair, str))
+        return list(filter(lambda market: market['currency'] == pair, self.get_markets()))
+
+    # List keepers open orders
+    def get_orders(self, pair: str) -> List[Order]:
+        assert (isinstance(pair, str))
+
+        currency = pair.split('-')[0]
+
+        orders = self._http_authenticated_request("POST", f"/v2/order/limit_orders/", {"currency": currency})
+        return list(map(lambda item: CoinoneOrder.from_message(item, pair), orders["limitOrders"]))
+
+    # Calculate the asking price units for a given price range
+    def _calc_price_precision(self, price: float) -> int:
+        if 1000 <= price < 5000:
+            return 1
+        elif 5000 <= price < 10000:
+            return 5
+        elif 10000 <= price < 50000:
+            return 10
+        elif 50000 <= price < 100000:
+            return 50
+        elif 100000 <= price < 500000:
+            return 100
+        elif 500000 <= price < 1000000:
+            return 500
+        elif 1000000 <= price:
+            return 1000
```

Comment:
Should we throw a runtime exception here if price is under 1000 or over 1000000?

Reply:
Good point, adding the remaining price ranges now